### DISCUSSION TITLE
How to add images to JSON dataset?

### DISCUSSION THREAD

**Discussion by DavidEdwardDarden** [nss-day-cohort-47/C47-Discussions#83] [Help Me]
I added images to a folder in src/componenets. I also added an image directly into the section I am working on (friends). So in FriendCard.js I have:

 
    //     <img src={friend.user.image} alt="person"></img>
    //     <img src="./nutshell.jpg" alt="person2"></img>

My dataset has:
  // "users": [
  //   {
  //     "email": "davidedarden83@gmail.com",
  //     "name": "David Darden",
  //     "id": 1,
  //     "image": "./nutshell.jpg",
  //     "bio": "I am focused on web development, data analysis, and data visualization. I have (..) as a trusted advisor to the customer’s technology needs. C# JavaScript React HTML CSS "
  //   },

I tried all the different versions of ./nutshell.jpg.... nutshell.jpg... ../nutshell.jpg ../../nutshell.jpg. I tried pulling an image from the images folder... and I tried putting the image directly in the friends folder. I also tried different versions of the file paths both in my JSON file and in my Friend Card. I have googled "put image in json file react" and the results have been inconclusive and I have not been able to get an image to show up as part of my Friend Card.

I also tried:
https://websemantics.uk/tools/image-to-data-uri-converter/
which is not a good option after seeing how many lines of data an image holds

**Comment by BryanNilsen** [comment]
I think you may have figured this out already.
If so, it may be helpful to document how you got your images working. There are several approaches and I'm curious as to which you got working.

**Reply by DavidEdwardDarden** [reply]
I'm not 100% about everything regarding this issue... but... here are the notes I wrote down from what the instructors said.

You could import your image and in the card, do something like this (nutshell.jpg was my file name) : 
`<img src={nutshell} alt="person"></img>`

You could mess around with putting your image in the public portion of your file structure.

You could get your image from an external API or have your images hosted by an online tool or service or something. (I'm not sure how this works with images.)

The following code successfully connected my friend card to my JSON file:
(I cannot get this code to post correctly because it has back tiks in it and that's how you insert code.)
`<img src= {require(`../images/${friend.user.image}`).default} alt="person"></img>`

I haven't Googled it yet but that "require" and ".default" are some of the key aspects to making this code work.

**Reply by BryanNilsen** [reply]
From Stack Overflow: [https://stackoverflow.com/a/62262850](https://stackoverflow.com/a/62262850)

Require returns an object and image's path can be ejected from its default property:
`<img src={require('path/to/img.png').default} />`

All of this really depends on where you're storing your images and how you're accessing them within your application. There are so many variables here, it's hard to pin down one way to do it.

You also have to take into consideration whether the images are loading dynamically based off of object properties. You will have to account for the initial render, where the path to the image is not defined... you can predefine a default image in the initialized value from state OR conditionally render the image if the property does/doesn't exist.

Isn't this fun?!?!?
